hubspot/connector.go

```diff
@@ -1,57 +1,45 @@
 package hubspot
 
 import (
+	"github.com/amp-labs/connectors"
 	"github.com/amp-labs/connectors/common"
-	"github.com/amp-labs/connectors/providers"
+	"github.com/amp-labs/connectors/proxy"
 )
 
-// Connector is a Hubspot connector.
+// Option is a function which mutates the hubspot connector configuration.
+type Option func(params *Connector)
+
+// Connector is a Hubspot connector that wraps around a proxy connector
+// and extends it with Hubspot-specific functionality (read, write, etc).
 type Connector struct {
-	Module  string
-	BaseURL string
-	Client  *common.JSONHTTPClient
+	*proxy.Connector
 }
 
-// NewConnector returns a new Hubspot connector.
+// NewConnector returns a new hubspot connector.
 func NewConnector(opts ...Option) (conn *Connector, outErr error) {
-	defer func() {
-		if re := recover(); re != nil {
-			tmp, ok := re.(error)
-			if !ok {
-				panic(re)
-			}
-
-			outErr = tmp
-			conn = nil
-		}
-	}()
+	conn = &Connector{}
 
-	params := &hubspotParams{}
 	for _, opt := range opts {
-		opt(params)
+		opt(conn)
 	}
 
-	var err error
-
-	params, err = params.prepare()
-	if err != nil {
-		return nil, err
-	}
+	if conn.Connector == nil {
+		conn = nil
+		outErr = common.ErrNoProxyConnectorFound
 
-	// Read provider info & replace catalog variables with given substitutions, if any
-	providerInfo, err := providers.ReadInfo(providers.Hubspot, nil)
-	if err != nil {
-		return nil, err
+		return
 	}
 
-	params.client.HTTPClient.Base = providerInfo.BaseURL
-	conn = &Connector{
-		BaseURL: params.client.HTTPClient.Base,
-		Module:  params.module,
-		Client:  params.client,
-	}
+	return
+}
 
-	conn.Client.HTTPClient.ErrorHandler = conn.interpretError
+func WithProxyConnector(conn connectors.ProxyConnector) Option {
+	return func(connector *Connector) {
+		proxyConn, ok := conn.(*proxy.Connector)
+		if !ok {
+			return
```

Comment:
Honestly, I'd panic here. There's code (the recover block) which will catch this panic and do the right thing. You don't want to silently fail.